### PR TITLE
Insertion sort

### PR DESCRIPTION
**Pull request by johnsonjew**

### CODE REVIEW

File: insertion_sort.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        value = list[index]
+        i = index - 1
+        while i >= 0:
+            if value < list[i]:
+                list[i + 1] = list[i]
+                list[i] = value
+                i = i - 1
+            else:
+                break
+
```
Comment (joelstanner) [inline insertion_sort.py:16]: Avoid using python reserved keywords as variables like the plague (list). This is a a doorway to bugs from hell.

File: insertion_sort.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+if __name__ == '__main__':
+    def worst_case():
+        list_a = range(100)
+        list_a.reverse()
+        return list_a
+
+    def best_case():
+        list_b = range(100)
+        return list_b
+
```
Comment (joelstanner) [inline insertion_sort.py:27]: There needs to be a variety of list lengths here. From the assignment description: "demonstrates the performance characteristics of this sort _over a variety of lengths of input_ in both the best and worst-case scenarios."

File: test_insert.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    sort_reverse_list = [15, 10, 9, 8, 7, 5, 2, 0]
+    insertion_sort(reverse_list)
+    float_list = [10, 7, 8, 9.5, 2.5, 15, 0, 5]
+    sort_float_list = [10, 7, 8, 9.5, 2.5, 15, 0, 5]
+    insertion_sort(float_list)
+    assert simple_list == sorted(sort_simple_list)
+    assert repeat_list == sorted(sort_repeat_list)
+    assert already_sorted == sorted(sort_already_sorted)
+    assert reverse_list == sorted(sort_reverse_list)
+    assert float_list == sorted(sort_float_list)
```
Comment (joelstanner) [inline test_insert.py:25]: This test needs to be broken into multiple tests - one or two asserts are usually the max for unit tests. It's not easy to see exactly what is being tested here at a glance. Also consider testing edge cases such as lists with one or zero items.
